### PR TITLE
chore: fix cyclic dependencies

### PR DESCRIPTION
**Pull request by ChristianBusshoff**
- moved @sit-onyx/storybook-utils and @sit-onyx/eslint-plugin from root to the repos where they are directly used
- chore(storybook-utils): Renamed internal `index.css`, which is already exported as `styles.css` to have the same name internally
- chore(storybook-utils): Removed `sit-onyx` dependency:
  - Load colors from DOM using `getComputedStyle(el).getPropertyValue(property)`
  - Removed `brand` defaults for `createTheme` to keep users from using our brand details
  - Moved `ONYX_BREAKPOINTS` constant and `OnyxBreakpoint` type to `@sit-onyx/shared`
    - `@sit-onyx/shared` is now published as we have direct build and runtime dependencies to it
    -  `turbo` configuration has been updated, so that it builds shared beforehand
 - updated `turbo` to determine dependencies itself
   - duplicated `peerDependencies` from the *workspace* into `devDependencies` as `turbo` will not consider them otherwise (See also: https://github.com/vercel/turborepo/commit/1eb3333f048ddc246caafbde920dd122e66b7e87)
- updated and added override for `sass-embedded` to fix incompatible vite plugin types (different version usages of `sass-embedded` will result in dedicated `vite` dependencies and therefore inequal types)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "publint:all": "pnpm -r --parallel --aggregate-output exec publint",
     "prepare": "simple-git-hooks",
     "gh:show-report": "./scripts/show-last-test-report.sh",
-    "preinstall": "npx only-allow pnpm"
+    "preinstall": "npx only-allow pnpm && pnpm list --depth=-1 --long | grep cycle && echo 'Cyclic dependency detected!' && exit 1 || echo 'No cyclic dependencies'"
```
Comment (JoCa96) [inline package.json:19]: `pnpm list --depth=-1 --long` doesn't output anything for me, can you check the script please?

**Review by JoCa96 — CHANGES_REQUESTED** [review]

File: packages/shared/tsconfig.json
Comment (larsrickert) [inline packages/shared/tsconfig.json:1]: Would prefer to use the same tsconfig that we use for the other packages too:

```json
{
  "extends": "@tsconfig/node22/tsconfig.json",
  "include": ["src"],
  "exclude": ["src/**/*.ct.*", "src/**/*.spec.*"],
  "compilerOptions": {
    "composite": true,
    "rootDir": "./src",
    "baseUrl": ".",
    "outDir": "dist",
    "declaration": true,
    "types": ["node"]
  }
}
```

Comment (larsrickert) [reply packages/shared/tsconfig.json:1]: Or at least remove all the uncommented code

File: packages/sit-onyx/package.json [hunk 2]
```diff
@@ -34,6 +34,7 @@
     "url": "https://github.com/SchwarzIT/onyx/issues"
   },
   "scripts": {
+    "td": "turbo watch --filter=sit-onyx dev",
```
Comment (larsrickert) [inline packages/sit-onyx/package.json:37]: What is this script and when should it be used?

File: packages/shared/src/index.ts
Comment (larsrickert) [inline packages/shared/src/index.ts:1]: Is this empty file intended?

File: packages/icons/package.json [hunk 3]
```diff
@@ -22,6 +22,7 @@
     "./*": "./src/assets/*"
   },
   "scripts": {
+    "build": "tsc",
```
Comment (larsrickert) [inline packages/icons/package.json:25]: Whats the purpose of this? The dist folder is not published in the npm package

File: packages/shared/src/breakpoints.ts
Comment (larsrickert) [inline packages/shared/src/breakpoints.ts:1]: Should we re-export the breakpoints in sit-onyx so the project can use them for individual logic without needing to install @sit-onyx/shared?

File: packages/sit-onyx/src/styles/grid.ct.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable playwright/expect-expect */
 import type { Locator, Page } from "@playwright/test";
+import { ONYX_BREAKPOINTS, type OnyxBreakpoint } from "@sit-onyx/shared/breakpoints";
```
Comment (larsrickert) [inline packages/sit-onyx/src/styles/grid.ct.tsx:3]: Why is there a dedicated `/breakpoints` path?
Shouldn't we just export all relevant exports directly from `@sit-onyx/shared`?

File: packages/storybook-utils/src/preview.ts [hunk 5]
```diff
@@ -40,7 +35,15 @@ const themes = {
  * export default preview;
  * ```
  */
-export const createPreview = <T extends Preview = Preview>(overrides?: T) => {
+export const createPreview = <T extends Preview = Preview>(
+  brandDetails: BrandDetails,
+  overrides?: T,
```
Comment (larsrickert) [inline packages/storybook-utils/src/preview.ts:40]: Would prefer to make the brandDetails optional as second parameter since most projects probably don't care about this

File: packages/storybook-utils/src/style.css
Comment (larsrickert) [inline packages/storybook-utils/src/style.css:1]: Why did you remove the export? onyx CSS variables are used in this file so we need to import them :)
The storybook utils are independent from the onyx component library so you could use the onyx theme without setting up sit-onyx itself so we can not rely on that the variables are imported in other places

File: packages/storybook-utils/src/theme.ts
Comment (larsrickert) [inline packages/storybook-utils/src/theme.ts:1]: If the `sit-onyx/themes/onyx.json` is unused now you can delete the file and also adjust `.github/workflows/import-figma.yml` to not import it any longer since it was only needed/used here

File: packages/storybook-utils/package.json [hunk 6]
```diff
@@ -28,16 +28,18 @@
...
     "storybook-dark-mode": ">= 4"
   },
   "dependencies": {
     "deepmerge-ts": "^7.1.3"
   },
   "devDependencies": {
-    "vue": "catalog:"
+    "vue": "catalog:",
+    "@sit-onyx/shared": "workspace:^",
+    "@sit-onyx/icons": "workspace:^"
```
Comment (larsrickert) [inline packages/storybook-utils/package.json:43]:
```suggestion
    "vue": "catalog:"
```

These two are already peerDependencies so they can be removed here

File: packages/storybook-utils/src/theme.ts [hunk 7]
```diff
@@ -1,33 +1,40 @@
...
  * Creates a custom theme for Storybook that uses onyx colors.
  *
  * @see https://storybook.js.org/docs/react/configure/theming#create-a-theme-quickstart
  */
-export const createTheme = (
-  options?: Pick<ThemeVarsPartial, "base" | "brandTitle" | "brandImage" | "brandUrl">,
-) => {
-  const base = options?.base ?? "light";
-  const primaryColor = onyxVariables["onyx-color-themed-primary-500"];
+export const createTheme = (brandDetails: BrandDetails, base: "light" | "dark" = "light") => {
```
Comment (larsrickert) [inline packages/storybook-utils/src/theme.ts:21]: Why is base not included in the brandDetails?

File: packages/vitepress-theme/package.json [hunk 8]
```diff
@@ -34,5 +34,8 @@
   "dependencies": {
     "@fontsource-variable/source-code-pro": "^5.1.0",
     "@fontsource-variable/source-sans-3": "^5.1.0"
+  },
+  "devDependencies": {
+    "sit-onyx": "workspace:^"
```
Comment (larsrickert) [inline packages/vitepress-theme/package.json:39]: onyx is already a peerDependency, why is it added as devDependency as well?

File: turbo.json [hunk 9]
```diff
@@ -20,10 +16,18 @@
     "test:components": {
       "outputs": ["playwright-report", "test-results"]
     },
+    "sit-onyx#test:components": {
+      "dependsOn": ["sit-onyx#build"]
+    },
```
Comment (larsrickert) [inline turbo.json:21]: Why are the tests dependent on the build? Playwright builds the components separately

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/package.json (hunk 2 above)
Comment (JoCa96) [reply packages/sit-onyx/package.json:37]: This script watches all dependencies of `sit-onyx#dev` and runs them

File: turbo.json (hunk 9 above)
Comment (JoCa96) [reply turbo.json:21]: You are right, what I wanted to achieve was "tests depend on the build of dependencies". Should be fixed thorugh the last commit.

File: packages/icons/package.json (hunk 3 above)
Comment (JoCa96) [reply packages/icons/package.json:25]: Done

File: packages/shared/src/breakpoints.ts
Comment (JoCa96) [reply packages/shared/src/breakpoints.ts:1]: good idea, done

File: packages/vitepress-theme/package.json (hunk 8 above)
Comment (JoCa96) [reply packages/vitepress-theme/package.json:39]: - updated turbo to determine dependencies itself

    -  duplicated peerDependencies from the workspace into devDependencies as turbo will not consider them otherwise (See also: vercel/turborepo@1eb3333)

File: packages/storybook-utils/package.json (hunk 6 above)
Comment (JoCa96) [reply packages/storybook-utils/package.json:43]: updated turbo to determine dependencies itself

    duplicated peerDependencies from the workspace into devDependencies as turbo will not consider them otherwise (See also: vercel/turborepo@1eb3333)

File: packages/sit-onyx/src/styles/grid.ct.tsx (hunk 4 above)
Comment (JoCa96) [reply packages/sit-onyx/src/styles/grid.ct.tsx:3]: The two exports to be bundled together, as they are address different run targets

File: packages/storybook-utils/src/preview.ts (hunk 5 above)
Comment (JoCa96) [reply packages/storybook-utils/src/preview.ts:40]: No, because if anybody uses it should not default to our details.
This will create the impression that it is an official `onyx` product.

File: packages/shared/tsconfig.json
Comment (JoCa96) [reply packages/shared/tsconfig.json:1]: done

File: packages/shared/src/index.ts
Comment (JoCa96) [reply packages/shared/src/index.ts:1]: yes, otherwise the bundler complains

File: packages/storybook-utils/src/style.css
Comment (JoCa96) [reply packages/storybook-utils/src/style.css:1]: Okay, then I need to move this to the `sit-onyx` package to be able to remove the dependency

File: packages/storybook-utils/src/theme.ts (hunk 7 above)
Comment (JoCa96) [reply packages/storybook-utils/src/theme.ts:21]: Because the base is not related to the brand.

File: packages/storybook-utils/src/preview.ts (hunk 5 above)
Comment (larsrickert) [reply packages/storybook-utils/src/preview.ts:40]: I meant to not provide any defaults but make the parameter optional.
If not provided, it will fall back to the default Storybook logo. We shouldn't force projects to pass any brand details, there might not be any

File: packages/storybook-utils/src/theme.ts
Comment (JoCa96) [reply packages/storybook-utils/src/theme.ts:1]: Done

File: packages/storybook-utils/src/style.css
Comment (JoCa96) [reply packages/storybook-utils/src/style.css:1]: Okay, I was able to do use a relative import

File: packages/sit-onyx/src/index.ts [hunk 10]
```diff
@@ -1,5 +1,7 @@
 import "./styles/index.scss";
 
+export { ONYX_BREAKPOINTS } from "@sit-onyx/shared/breakpoints";
```
Comment (larsrickert) [inline packages/sit-onyx/src/index.ts:3]:
```suggestion
export { ONYX_BREAKPOINTS, type OnyxBreakpoint } from "@sit-onyx/shared/breakpoints";
```

File: packages/sit-onyx/src/styles/grid.ct.tsx (hunk 4 above)
Comment (larsrickert) [reply packages/sit-onyx/src/styles/grid.ct.tsx:3]: Should we then instead use a `@sit-onyx/shared/node` or `@sit-onyx/shared/browser` path so we don't have to create a new path in the future for every new feature? With the current name its limited to only include breakpoints

File: packages/storybook-utils/src/style.css [hunk 11]
```diff
@@ -1,4 +1,4 @@
-@import url("sit-onyx/themes/onyx.css");
+@import url("../../sit-onyx/src/styles/variables/themes/onyx.css");
```
Comment (larsrickert) [inline packages/storybook-utils/src/style.css:1]: This works in our local monorepo but will break when used in a real project since the relative path is not resolvable then...we need to import from `sit-onyx` instead

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/styles/grid.ct.tsx (hunk 4 above)
Comment (JoCa96) [reply packages/sit-onyx/src/styles/grid.ct.tsx:3]: Ideally, yes. But looking at the PR, its age and scope this is too much effort for now. Fo-Up was created: [#2056](https://github.com/SchwarzIT/onyx/issues/2056)

File: packages/storybook-utils/src/style.css [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+.sb-previewBlock,
+.sb-argstableBlock {
+  display: none;
+}
+
+/* 
+  Copy of ../sit-onyx/src/styles/variables/themes/onyx.css
+  TODO: Find a way to automate this
+ */
+:where(:root),
```
Comment (larsrickert) [inline packages/storybook-utils/src/style.css:41]: Hmm I don't really like this copy...what do you think about just replacing / hard-coding the two variables in line 8 and 23 since we only need those two values?

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/storybook-utils/src/preview.ts (hunk 5 above)
Comment (JoCa96) [reply packages/storybook-utils/src/preview.ts:40]: done

File: packages/storybook-utils/src/style.css (hunk 12 above)
Comment (larsrickert) [reply packages/storybook-utils/src/style.css:41]: Oh, you‘re right
